BaseStation/MainWindow/MapTileDownloadManager.cs

```diff
@@ -24,89 +25,122 @@ public class Configuration
             private const double MaxLongitude = 180;
 
             private int _scale;
+            private int _zoom;
+            private string _mapType;
+            private double _latitude;
+            private double _longitude;
+            private int _imgWidth;
+            private int _imgheight;
+            private int _tilingWidth;
+            private int _tilingHeight;
+
             public int Scale
             {
                 get { return _scale; }
-                set { if (value == 1 || value == 2) _scale = value; }
+                set {
+                    if (value == 1 || value == 2) _scale = value;
+                }
             }
-            private int _zoom;
             public int Zoom
             {
                 get { return _zoom; }
-                set { if (value >= 0 && value <= 21) _zoom = value; }
+                set {
+                    if (value >= 0 && value <= 21) _zoom = value;
+                }
             }
-            private string _mapType;
             public string MapType
             {
                 get { return _mapType; }
                 set
                 {
+                    value = value.ToLower();
                     if (value.Equals("roadmap") || value.Equals("satellite") ||
                       value.Equals("terrain") || value.Equals("hybrid")) _mapType = value;
                 }
             }
-            private Tuple<int, int> _imgDim;
-            public Tuple<int, int> ImgDim
-            {
-                get { return _imgDim; }
-                set { if (value.Item1 > 0 && value.Item2 > 0) _imgDim = value; }
+            public double Latitude { get { return _latitude; }
+                set {
+                    if (value >= MinLatitude && value <= MaxLatitude) _latitude = value;
```

Comment:
How would the error get handled when it's triggered? right now what happens is if someone tries to change it to an invalid value then no change occurs.

Reply:
The input validation should be happening in the UI (.xaml.cs) rather than in the model. That way, the model stays valid at all times, and any programmer mistakes where the wrong value is trying to be stored are made visible.